src/components/tripDetails/_FilterPopUp.svelte

```diff
@@ -0,0 +1,162 @@
+<script lang="ts">
+	import { goto } from '$app/navigation';
+	import { page } from '$app/stores';
+	import type { TravelData } from '$tripDomain';
+	import { onMount } from 'svelte';
+
+	export let trip: TravelData;
+
+	const endDateTrip = trip.endDate.toISOString().slice(0, 10);
+	const startDateTrip = trip.startDate.toISOString().slice(0, 10);
+
+	interface FilterOptions {
+		name: string;
+		queryParam: string;
+		value: string;
+		checked: boolean;
+		minValue: string;
+		maxValue: string;
+		isMin: boolean;
+	}
+
+	let filterOptions: Array<FilterOptions> = [
+		{
+			name: 'Minimal Start Date',
+			queryParam: 'minDeductionDate',
+			value: startDateTrip,
+			checked: false,
+			minValue: startDateTrip,
+			maxValue: endDateTrip,
+			isMin: true
+		},
+		{
+			name: 'Maximal Start Date',
+			queryParam: 'maxDeductionDate',
+			value: endDateTrip,
+			checked: false,
+			minValue: startDateTrip,
+			maxValue: endDateTrip,
+			isMin: false
+		},
+		{
+			name: 'Minimal End Date',
+			queryParam: 'minEndDate',
+			value: startDateTrip,
+			checked: false,
+			minValue: startDateTrip,
+			maxValue: endDateTrip,
+			isMin: true
+		},
+		{
+			name: 'Maximal End Date',
+			queryParam: 'maxEndDate',
+			value: endDateTrip,
+			checked: false,
+			minValue: startDateTrip,
+			maxValue: endDateTrip,
+			isMin: false
+		}
+	];
+
+	onMount(() => {
+		filterOptions = filterOptions.map((option) => {
+			return {
+				...option,
+				checked: $page.url.searchParams.has(option.queryParam),
+				value: $page.url.searchParams.get(option.queryParam) ?? option.value
+			};
+		});
+	});
+
+	function setQueryParameter(url: URL, parameter: string, value: string): URL {
+		if (url.searchParams.has(parameter)) {
+			url.searchParams.delete(parameter);
+		}
+		url.searchParams.append(parameter, value);
+		return url;
+	}
+
+	function changeDate(e: any, index: number) {
+		filterOptions[index] = { ...filterOptions[index], value: e.target.value };
+		console.log(filterOptions[index]);
+	}
+
+	function onChangeCheck(e: any, index: number) {
+		filterOptions[index] = { ...filterOptions[index], checked: e.target.checked };
+		console.log(filterOptions[index]);
+	}
+
+	function clearFilter() {
+		let url = new URL($page.url);
+		filterOptions.forEach((option) => {
+			url.searchParams.delete(option.queryParam);
+		});
+		goto(`?${url.searchParams.toString()}`);
```

Comment:
Wie `teuer` ist dieser Aufruf? Ist das n hard redirect oder ist das so schnell, dass mans nicht merkt?

Reply:
Das ist sehr schnell. Also so wie ich es verstanden habe machst du redirects in der load function serverseitig und goto im Browser clientside